<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Загрузка-данных" data-toc-modified-id="Загрузка-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Загрузка данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Обработка-пропусков,-неккоректных-значений" data-toc-modified-id="Обработка-пропусков,-неккоректных-значений-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Обработка пропусков, неккоректных значений</a></span></li><li><span><a href="#Замена-типа-данных" data-toc-modified-id="Замена-типа-данных-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Замена типа данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Лемматизация" data-toc-modified-id="Лемматизация-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Лемматизация</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Категоризация данных</a></span></li></ul></li><li><span><a href="#Исследование-влияния-параметров" data-toc-modified-id="Исследование-влияния-параметров-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Исследование влияния параметров</a></span><ul class="toc-item"><li><span><a href="#Зависимость-между-наличием-детей-и-возвратом-кредита-в-срок" data-toc-modified-id="Зависимость-между-наличием-детей-и-возвратом-кредита-в-срок-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Зависимость между наличием детей и возвратом кредита в срок</a></span></li><li><span><a href="#Зависимость-между-семейным-положением-и-возвратом-кредита-в-срок" data-toc-modified-id="Зависимость-между-семейным-положением-и-возвратом-кредита-в-срок-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Зависимость между семейным положением и возвратом кредита в срок</a></span></li><li><span><a href="#Зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок" data-toc-modified-id="Зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Зависимость между уровнем дохода и возвратом кредита в срок</a></span></li><li><span><a href="#Влияние-целей-кредита-на-возврат-его-в-срок" data-toc-modified-id="Влияние-целей-кредита-на-возврат-его-в-срок-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Влияние целей кредита на возврат его в срок</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

# Исследование надёжности заёмщиков

Для нужд кредитного отдела банка необходимо понять, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок.

## Загрузка данных

In [1]:
# Импорт библиотек
import pandas as pd
from pymystem3 import Mystem

Отключение вывода в прокручиваемом окне

In [2]:
%%javascript
IPython.OutputArea.prototype._should_scroll = function(lines) {
    return false;
}

<IPython.core.display.Javascript object>

In [3]:
# Загрузка данных
df = pd.read_csv('/datasets/data.csv')

# Ознакомление с данными
print(df.columns)
print()
print(df.info())
display(df.sample(5))
display(df.describe().T)

Index(['children', 'days_employed', 'dob_years', 'education', 'education_id',
       'family_status', 'family_status_id', 'gender', 'income_type', 'debt',
       'total_income', 'purpose'],
      dtype='object')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), 

children  days_employed  dob_years education  education_id  \
7259          0  -11556.539021         57    высшее             0   
20889         2   -3919.449417         40   среднее             1   
654           1    -997.705115         43    высшее             0   
11484         1    -124.051493         59    высшее             0   
1485          2            NaN         37   среднее             1   

               family_status  family_status_id gender income_type  debt  \
7259         женат / замужем                 0      F   сотрудник     0   
20889  Не женат / не замужем                 4      M   компаньон     0   
654          женат / замужем                 0      F   компаньон     0   
11484        женат / замужем                 0      F   компаньон     0   
1485         женат / замужем                 0      M   сотрудник     0   

        total_income                     purpose  
7259   201414.906802                недвижимость  
20889  133362.300391    операции с недвижимостью  
654    161238.916104                недвижимость  
11484  163612.577526  покупка жилой недвижимости  
1485             NaN             свой автомобиль

count           mean            std           min  \
children          21525.0       0.538908       1.381587     -1.000000   
days_employed     19351.0   63046.497661  140827.311974 -18388.949901   
dob_years         21525.0      43.293380      12.574584      0.000000   
education_id      21525.0       0.817236       0.548138      0.000000   
family_status_id  21525.0       0.972544       1.420324      0.000000   
debt              21525.0       0.080883       0.272661      0.000000   
total_income      19351.0  167422.302208  102971.566448  20667.263793   

                            25%            50%            75%           max  
children               0.000000       0.000000       1.000000  2.000000e+01  
days_employed      -2747.423625   -1203.369529    -291.095954  4.017554e+05  
dob_years             33.000000      42.000000      53.000000  7.500000e+01  
education_id           1.000000       1.000000       1.000000  4.000000e+00  
family_status_id       0.000000       0.000000       1.000000  4.000000e+00  
debt                   0.000000       0.000000       0.000000  1.000000e+00  
total_income      103053.152913  145017.937533  203435.067663  2.265604e+06

**Вывод**

В таблице 21525 строк.
Названия столбоц в порядке.
В данных есть проблемные значения, пропуски и некоректные типы данных.

## Предобработка данных

### Обработка пропусков, неккоректных значений

В столбце children имеются отрицательные значения -1. Отрицательными они быть не могут. Вероятно минус появился из-за ошибки при выгрузке.

In [4]:
df['children'] = df['children'].abs()

В столбце education строки отличаются регистром. Нужно привести всё к нижнему регистру.

In [5]:
df['education'] = df['education'].str.lower()

В столбце days_employed значения отработанных дней со знаком минус. Вероятно ошибка при выгрузке. Необходимо сделать все значения положительными.

В столбце days_employed присутствуют огромные значения, которые явно не являются количеством отработанных дней. Используя фильтр print(df[df['days_employed'] >= 20000].count()), определил, что после 20 000 дней количество строк 3445 не изменяется. Это значит, что на этом реальные значения закончились, а дальше начиная с сотен тысяч (приблизительно с 330 000) начнутся подозрительные. Вероятно, это ошибка при выгрузке, из-за которой значения стали на 2 порядка выше. Нужно это исправить.

In [6]:
df['days_employed'] = df['days_employed'].abs()

df.loc[df['days_employed'] > 20000, 'days_employed'] = df['days_employed'] / 100

В столбце dob_years присутствует 101 значение 0. Эту ошибку можно удалить из-за небольшого количества таких строк.

In [7]:
df = df.loc[df['dob_years'] != 0]

В столбце 'income_type' подсчитав количество каждых категорий, обнаружил, что 'безработный', 'предприниматель', 'в декрете', 'студент' в сумме всего лишь 6 значений. Этого не достаточно для анализа, поэтому эти строки можно удалить:

In [8]:
count_income_type = df['income_type'].value_counts()
df = df.loc[df['income_type'] != 'безработный']
df = df.loc[df['income_type'] != 'предприниматель']
df = df.loc[df['income_type'] != 'в декрете']
df = df.loc[df['income_type'] != 'студент']

В столбцах 'days_employed' и 'total_income' имеются пропуски NaN одновременно в обоих столбцах. Используя bool_series = pd.isnull(df['days_employed'])df_nan = df[bool_series] осмотрел строки с пропусками. Никаких закономерностей появления NaN по цели кредита 'purpose' не обнаружил.

В столбце 'income_type' пропуски отсутствуют у 'безработный', 'в декрете', 'студент'. Возможно пропуски появились во время выгрузки данных. Т.к. пропуски находятся в количественных переменных нужно заменить их медианными значениями. Среднее значение не подходит, т.к. немногочисленные строчки с большими значениями сильно оттянут среднее значение в свою сторону.

In [9]:
# Т.к. медиана находится в упорядоченном списке без пропусков удаляем в отдельном df все пропуски:
df_not_nan = df.dropna()
# Затем находим медианы для оставшихся 4 категорий:
goss_median_days_employed = df_not_nan[df_not_nan['income_type'] == 'госслужащий']['days_employed'].sort_values().median()
komp_median_days_employed = df_not_nan[df_not_nan['income_type'] == 'компаньон']['days_employed'].sort_values().median()
pens_median_days_employed = df_not_nan[df_not_nan['income_type'] == 'пенсионер']['days_employed'].sort_values().median()
sotr_median_days_employed = df_not_nan[df_not_nan['income_type'] == 'сотрудник']['days_employed'].sort_values().median()

goss_median_total_income = df_not_nan[df_not_nan['income_type'] == 'госслужащий']['total_income'].sort_values().median()
komp_median_total_income = df_not_nan[df_not_nan['income_type'] == 'компаньон']['total_income'].sort_values().median()
pens_median_total_income = df_not_nan[df_not_nan['income_type'] == 'пенсионер']['total_income'].sort_values().median()
sotr_median_total_income = df_not_nan[df_not_nan['income_type'] == 'сотрудник']['total_income'].sort_values().median()

# Составляем дополнительный DataFrame с средними значениями:
data = [['госслужащий', 2673.404955519986, 150521.9175795368],
        ['компаньон', 1548.6375436404599, 172280.58171182536],
        ['пенсионер', 3651.763367751465, 118557.17941467627],
        ['сотрудник', 1576.0676889889892, 142594.39684740017]]
columns = ['income_type', 'days_employed', 'total_income']
df_2 = pd.DataFrame(data=data, columns=columns)
# Cоединяем основной и дополнительный DF с сохранением всех значений основного:
df = df.merge(df_2, on='income_type', how='left')
# Заменяем пропуски на соответствущие средние значения:
df['days_employed_x'] = df['days_employed_x'].fillna(df['days_employed_y'])
df['total_income_x'] = df['total_income_x'].fillna(df['total_income_y'])
# Теперь необходимо удалить вспомогательные столбцы, они больше не нужны:
df = df.drop(columns='days_employed_y')
df = df.drop(columns='total_income_y')

В столбце 'children' имеется 75 значений с количеством детей равным 20. Это ошибка, возможно вызвана опечаткой. Изменим количество детей на 2.

In [10]:
df['children'] = df['children'].replace(20, 2)

**Вывод**

В данных исправлены ошибки, такие как: отрицательные значения опыта работы и количества детей, разные регистры в столбце образования, огромные значения опыта работы, нулевые значения возраста и пропущенные значения в столбцах отработанных дней и общего дохода. Удалил 6 строк с редко повторяющимися категориями. Данные подготовлены для дальнейшего исследования.

### Замена типа данных

In [11]:
df['days_employed_x'] = df['days_employed_x'].astype(int)
df['total_income_x'] = df['total_income_x'].astype(int)
df

children  days_employed_x  dob_years education  education_id  \
0             1             8437         42    высшее             0   
1             1             4024         36   среднее             1   
2             0             5623         33   среднее             1   
3             3             4124         32   среднее             1   
4             0             3402         53   среднее             1   
...         ...              ...        ...       ...           ...   
21413         1             4529         43   среднее             1   
21414         0             3439         67   среднее             1   
21415         1             2113         38   среднее             1   
21416         3             3112         38   среднее             1   
21417         2             1984         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем                 0      F   сотрудник     0   
2       женат / замужем                 0      M   сотрудник     0   
3       женат / замужем                 0      M   сотрудник     0   
4      гражданский брак                 1      F   пенсионер     0   
...                 ...               ...    ...         ...   ...   
21413  гражданский брак                 1      F   компаньон     0   
21414   женат / замужем                 0      F   пенсионер     0   
21415  гражданский брак                 1      M   сотрудник     1   
21416   женат / замужем                 0      M   сотрудник     1   
21417   женат / замужем                 0      F   сотрудник     0   

       total_income_x                       purpose  
0              253875                 покупка жилья  
1              112080       приобретение автомобиля  
2              145885                 покупка жилья  
3              267628    дополнительное образование  
4              158616               сыграть свадьбу  
...               ...                           ...  
21413          224791             операции с жильем  
21414          155999          сделка с автомобилем  
21415           89672                  недвижимость  
21416          244093  на покупку своего автомобиля  
21417           82047         на покупку автомобиля  

[21418 rows x 12 columns]

**Вывод**

Теперь в таблице целочисленные значения, которые легче визуально воспринимать.

### Обработка дубликатов

In [12]:
# Определение числа дубликатов
df.duplicated().sum()
# Удаление дубликатов
df.drop_duplicates().reset_index(drop=True)

children  days_employed_x  dob_years education  education_id  \
0             1             8437         42    высшее             0   
1             1             4024         36   среднее             1   
2             0             5623         33   среднее             1   
3             3             4124         32   среднее             1   
4             0             3402         53   среднее             1   
...         ...              ...        ...       ...           ...   
21342         1             4529         43   среднее             1   
21343         0             3439         67   среднее             1   
21344         1             2113         38   среднее             1   
21345         3             3112         38   среднее             1   
21346         2             1984         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем                 0      F   сотрудник     0   
2       женат / замужем                 0      M   сотрудник     0   
3       женат / замужем                 0      M   сотрудник     0   
4      гражданский брак                 1      F   пенсионер     0   
...                 ...               ...    ...         ...   ...   
21342  гражданский брак                 1      F   компаньон     0   
21343   женат / замужем                 0      F   пенсионер     0   
21344  гражданский брак                 1      M   сотрудник     1   
21345   женат / замужем                 0      M   сотрудник     1   
21346   женат / замужем                 0      F   сотрудник     0   

       total_income_x                       purpose  
0              253875                 покупка жилья  
1              112080       приобретение автомобиля  
2              145885                 покупка жилья  
3              267628    дополнительное образование  
4              158616               сыграть свадьбу  
...               ...                           ...  
21342          224791             операции с жильем  
21343          155999          сделка с автомобилем  
21344           89672                  недвижимость  
21345          244093  на покупку своего автомобиля  
21346           82047         на покупку автомобиля  

[21347 rows x 12 columns]

**Вывод**

В таблице были обнаружены 71 дубликат. Данные строки были удалены. Дубликаты могли появиться на этапе формирования данных. Так же дубликаты могли появиться на этапе выгрузки данных из базы.

### Лемматизация

В столбце с целями кредита находятся одинаковые по смыслу, но разные по написанию цели кредита. Необходимо унифицировать одинаковые цели кредита.

In [13]:
# Создадим переменную со всеми значениями целей получения кредита, но без повторов, для того, что бы ознакомиться с целями кредита.
column_purpose = df['purpose'].value_counts()
print(column_purpose)

# Найдём леммы в столбце 'purpose' и запишем их в новый столбце 'purpose_lemm'.
m = Mystem()

def func_lemm(string):
    return m.lemmatize(string)
df['purpose_lemm'] = df['purpose'].apply(func_lemm)
df.head(5)

# Создадим новый столбец purpose_x в котором будут указаны цели кредита одним словом
def lemm(word_lemm):
    if 'жилье' in word_lemm:
        return 'недвижимость'
    if 'недвижимость' in word_lemm:
        return 'недвижимость'
    if 'автомобиль' in word_lemm:
        return 'автомобиль'
    if 'свадьба' in word_lemm:
        return 'свадьба'
    if 'образование' in word_lemm:
        return 'образование'
df['purpose_x'] = df['purpose_lemm'].apply(lemm)

# Столбцы 'purpose' и 'purpose_lemm' можно удалить
df = df.drop(columns = 'purpose')
df = df.drop(columns = 'purpose_lemm')
df.head(5)

свадьба                                   792
на проведение свадьбы                     772
сыграть свадьбу                           769
операции с недвижимостью                  673
покупка коммерческой недвижимости         661
покупка жилья для сдачи                   650
операции с коммерческой недвижимостью     649
операции с жильем                         647
покупка жилья                             641
жилье                                     641
покупка жилья для семьи                   640
строительство собственной недвижимости    632
недвижимость                              630
операции со своей недвижимостью           630
строительство жилой недвижимости          623
покупка недвижимости                      621
строительство недвижимости                620
покупка своего жилья                      619
ремонт жилью                              609
покупка жилой недвижимости                603
на покупку своего автомобиля              502
заняться высшим образованием      

children  days_employed_x  dob_years education  education_id  \
0         1             8437         42    высшее             0   
1         1             4024         36   среднее             1   
2         0             5623         33   среднее             1   
3         3             4124         32   среднее             1   
4         0             3402         53   среднее             1   

      family_status  family_status_id gender income_type  debt  \
0   женат / замужем                 0      F   сотрудник     0   
1   женат / замужем                 0      F   сотрудник     0   
2   женат / замужем                 0      M   сотрудник     0   
3   женат / замужем                 0      M   сотрудник     0   
4  гражданский брак                 1      F   пенсионер     0   

   total_income_x     purpose_x  
0          253875  недвижимость  
1          112080    автомобиль  
2          145885  недвижимость  
3          267628   образование  
4          158616       свадьба

**Вывод**

Произведя лемматизацию по каждому столбцу, появляется возможность отследить повторяющиеся слова. Для лемматизации использовал специальную библиотеку pymystem3. Написав функцию для нахождения лемм в строке, указываем её через метод .apply() к столбцу 'purpose'. Создал новый столбце, в котором указаны цели кредита одним словом.

### Категоризация данных

Т.к. нужно определить как наличие детей, семейный статус, уровень дохода и цель кредита влияют на возврат его в срок, категоризацию буду проводить для этих параметров.

Категоризацию по количеству детей в семье проведу по значениям: нет детей, 1 ребёнок, 2 ребёнка 3 и более детей. 1, 2
и 3 объединять не буду т.к. важно проследить изменения с каждым последующим ребёнком. А вот 3, 4 и 5 объединю, т.к.
таких значений в общей сложности всего лишь 378. Словарь создам из столбцов 'category_children' и 'category_children_id'.

In [14]:
# Категоризация
def category_children(v):
    if v == 0:
        return 'нет детей'
    if v == 1:
        return '1 ребёнок'
    if v == 2:
        return '2 ребёнка'
    if v >= 3:
        return '3 и более детей'
    
df['category_children'] = df['children'].apply(category_children)

def category_children_id(b):
    if b == 'нет детей':
        return 0
    if b == '1 ребёнок':
        return 1
    if b == '2 ребёнка':
        return 2
    if b >= '3 и более детей':
        return 3
df['category_children_id'] = df['category_children'].apply(category_children_id)
category_children = df[['category_children', 'category_children_id']]
category_children = category_children.drop_duplicates().reset_index(drop=True)
category_children = category_children.sort_values('category_children_id')
display(category_children)

category_children  category_children_id
1         нет детей                     0
0         1 ребёнок                     1
3         2 ребёнка                     2
2   3 и более детей                     3

Столбец 'family_status' тоже является категоризацией состоящей из 5 значений: 'женат / замужем', 'гражданский брак', 'вдовец / вдова', 'в разводе', 'Не женат / не замужем'. Создам словарь из столбцов 'family_status', 'family_status_id'.

In [15]:
# Категоризация
family_status_dict = df[['family_status', 'family_status_id']]
family_status_dict = family_status_dict.drop_duplicates().reset_index(drop=True)
display(family_status_dict)

family_status  family_status_id
0        женат / замужем                 0
1       гражданский брак                 1
2         вдовец / вдова                 2
3              в разводе                 3
4  Не женат / не замужем                 4

По столбцу total_income_x категоризуем данные. Разобьём на следующие интервалы 50, 100, 150, 200, 400, 401+ тысяч рублей.

In [16]:
# Категоризация
def category_income(x):
    if x <= 50000:
        return 'до 50тр'
    if x <= 100000:
        return '50-100тр'
    if x <= 150000:
        return '100-150тр'
    if x <= 200000:
        return '150-200тр'
    if x <= 400000:
        return '200-400тр'
    return '400+ тр'
df['category_income'] = df['total_income_x'].apply(category_income)

def category_income_id(c):
    if c == 'до 50тр':
        return 0
    if c == '50-100тр':
        return 1
    if c == '100-150тр':
        return 2
    if c == '150-200тр':
        return 3
    if c == '200-400тр':
        return 4
    if c == '400+ тр':
        return 5
df['category_income_id'] = df['category_income'].apply(category_income_id)
category_income = df[['category_income', 'category_income_id']]
category_income = category_income.drop_duplicates().reset_index(drop=True)
category_income = category_income.sort_values('category_income_id')
display(category_income)

category_income  category_income_id
5         до 50тр                   0
3        50-100тр                   1
1       100-150тр                   2
2       150-200тр                   3
0       200-400тр                   4
4         400+ тр                   5

Категоризацию по цели кредиты выполнил на этапе лемматизации и получил следующие категории

In [17]:
# Категоризация
def purpose_id(z):
    if z == 'недвижимость':
        return 0
    if z == 'автомобиль':
        return 1
    if z == 'образование':
        return 2
    if z == 'свадьба':
        return 3
df['purpose_id'] = df['purpose_x'].apply(purpose_id)
purpose_dict = df[['purpose_x', 'purpose_id']]
purpose_dict = purpose_dict.drop_duplicates().reset_index(drop=True)
display(purpose_dict)

display(df.head(5))

purpose_x  purpose_id
0  недвижимость           0
1    автомобиль           1
2   образование           2
3       свадьба           3

children  days_employed_x  dob_years education  education_id  \
0         1             8437         42    высшее             0   
1         1             4024         36   среднее             1   
2         0             5623         33   среднее             1   
3         3             4124         32   среднее             1   
4         0             3402         53   среднее             1   

      family_status  family_status_id gender income_type  debt  \
0   женат / замужем                 0      F   сотрудник     0   
1   женат / замужем                 0      F   сотрудник     0   
2   женат / замужем                 0      M   сотрудник     0   
3   женат / замужем                 0      M   сотрудник     0   
4  гражданский брак                 1      F   пенсионер     0   

   total_income_x     purpose_x category_children  category_children_id  \
0          253875  недвижимость         1 ребёнок                     1   
1          112080    автомобиль         1 ребёнок                     1   
2          145885  недвижимость         нет детей                     0   
3          267628   образование   3 и более детей                     3   
4          158616       свадьба         нет детей                     0   

  category_income  category_income_id  purpose_id  
0       200-400тр                   4           0  
1       100-150тр                   2           1  
2       100-150тр                   2           0  
3       200-400тр                   4           2  
4       150-200тр                   3           3

**Вывод**

Произведена категоризация по количеству детей, семеному статусу, уровеню дохода и цели кредита. Созданы словари для тех же параметров.

## Исследование влияния параметров

### Зависимость между наличием детей и возвратом кредита в срок

In [18]:
# Расчёт показателей
children_final = df.pivot_table(index = ['category_children'], values = 'debt', aggfunc = 'mean')
children_final['%'] = children_final['debt'] * 100
# Вывод информации
children_final = children_final.reset_index()
children_final.columns = ['Наличие детей', 'Доля должников', 'Процент должников']
display(children_final.sort_values(by='Процент должников', ascending=False))
print('Число кредитов по категориям:')
print(df['category_children'].value_counts())

Наличие детей  Доля должников  Процент должников
1        2 ребёнка        0.094991           9.499055
0        1 ребёнок        0.090965           9.096535
2  3 и более детей        0.082011           8.201058
3        нет детей        0.075163           7.516340

Число кредитов по категориям:
нет детей          14076
1 ребёнок           4848
2 ребёнка           2116
3 и более детей      378
Name: category_children, dtype: int64


**Вывод**

Как видно из данных, меньше всего долгов среди тех, кто детей не имеет. А с ростом количества детей, растёт доля должников. Больше всего значений в категории "нет детей". Категория "3 и более детей" выбивается из общей тенденции возможно из-за малого количества случаев (всего 378).

Итак, с ростом количества детей, немного увеличивается риск невозврата кредита в срок.

### Зависимость между семейным положением и возвратом кредита в срок

In [19]:
# Расчёт показателей
family_status_final = df.pivot_table(index = ['family_status'], values = ['debt', 'dob_years', 'total_income_x'], aggfunc = 'mean')
family_status_final['%'] = family_status_final['debt'] * 100

# Вывод информации
family_status_final = family_status_final.reset_index()
family_status_final.columns = ['Семейный статус', 'Доля должников', 'Возраст клиента', 'Ежемесячный доход', 'Процент должников']
display(family_status_final.sort_values(by='Процент должников', ascending=False))
print('Число кредитов по категориям:')
display(df['family_status'].value_counts())

Семейный статус  Доля должников  Возраст клиента  Ежемесячный доход  \
0  Не женат / не замужем        0.097639        38.594778      166366.287196   
3       гражданский брак        0.092922        42.346413      164599.654068   
4        женат / замужем        0.075032        43.738076      166707.766223   
1              в разводе        0.071730        45.902110      167801.870886   
2         вдовец / вдова        0.064921        56.809424      142715.270157   

   Процент должников  
0           9.763948  
3           9.292248  
4           7.503245  
1           7.172996  
2           6.492147

Число кредитов по категориям:


женат / замужем          12328
гражданский брак          4154
Не женат / не замужем     2796
в разводе                 1185
вдовец / вдова             955
Name: family_status, dtype: int64

**Вывод**

Люди без пары чаще всего имеют долги по кредитам (9,76%), при этом люди, живущие в гражданском браке, реже имеют долги по кредитам (9,29%) и люди, живущие в браке ещё реже (7,5%). Это говорит о том, что люди, живущие в браке, более серьёзно подходят к процессу взятия кредита и рассчитывают свой бюджет. Они более предсказуемы и несут обязательства перед семьёй. Поэтому долги по кредитам ставят под удар не только их, но и всех членов семьи. Отсюда вытекает бОльшая надёжность по выплате в семейных людях. В категории "в разводе" процент должников 7,17%. Вероятно способность выплачивать кредит в срок связана с возрастом в этих группах. Они самые зрелые. При этом уровень дохода везде практически одинаковый.

Итак, люди, состоящие в браке больше остальных, способны платить за кредит в срок. При этом люди в разводе, так же способны платить за кредит из-за сохранения после развода их дохода (и вероятно разделения имущества). Группа вдова/вдовец самая возрастная и хоть доходы там меньше чем в других категориях, кредит они платят исправно. Это связано с тем, что в возрасте люди больше тратят на что-то практическое, чем на эмоциональное.

### Зависимость между уровнем дохода и возвратом кредита в срок

In [20]:
# Расчёт показателей
category_income_final = df.pivot_table(index = ['category_income'], values = ['debt'], aggfunc = 'mean')
category_income_final['%'] = category_income_final['debt'] * 100

# Вывод информации
category_income_final = category_income_final.reset_index()
category_income_final.columns = ['Доход', 'Доля должников', 'Процент должников']
display(category_income_final.sort_values(by='Процент должников', ascending=False))
print('Число кредитов по категориям:')
display(df['category_income'].value_counts())

Доход  Доля должников  Процент должников
0  100-150тр        0.086346           8.634594
1  150-200тр        0.084614           8.461377
4   50-100тр        0.080649           8.064913
2  200-400тр        0.072268           7.226779
5    до 50тр        0.062162           6.216216
3    400+ тр        0.058824           5.882353

Число кредитов по категориям:


100-150тр    7192
150-200тр    4751
200-400тр    4511
50-100тр     4067
400+ тр       527
до 50тр       370
Name: category_income, dtype: int64

**Вывод**

Среди категорий с большим числом кредитов самой надёжной является категория с доходами "200-400 тр". В целом по всем категориям процент должников примерно 8%. В категориях "400+ тр" и "до 50 тр" доля должников примерно 6% но и количество таких событий немногочисленно и возможно низкий процент связан с малым числом данных. 

Итак, между уровнем дохода и возвратом кредита в срок не наблюдается строгой и линейной зависимости.

### Влияние целей кредита на возврат его в срок

In [21]:
# Расчёт показателей
purpose_final = df.pivot_table(index = ['purpose_x'], values = ['debt'], aggfunc = 'mean')
purpose_final['%'] = purpose_final['debt'] * 100

# Вывод информации
purpose_final = purpose_final.reset_index()
purpose_final.columns = ['Цель кредита', 'Доля должников', 'Процент должников']
display(purpose_final.sort_values(by='Процент должников', ascending=False))

Цель кредита  Доля должников  Процент должников
0    автомобиль        0.092964           9.296365
2   образование        0.092408           9.240759
3       свадьба        0.078868           7.886841
1  недвижимость        0.072110           7.211048

**Вывод**

По целям кредита, очевидно, что самыми надёжными целями служат свадьба и недвижимость. Эти категории относятся к более возрастным людям, а так же связаны с построением семьи. А образование и автомобиль в свою очередь имеют выше долю задолженностей по кредитам. Эти категории относятся к менее возрастным людям.

Итак, семейные люди берущие кредит на благо семьи являются более надёжными заёмщиками.

## Общий вывод

В результате исследования получаем:

1) Между наличием детей и возвратом кредита в срок зависимость есть - с ростом числа детей, немного растёт доля должников.

2) Между семейным положением и возвратом кредита в срок зависимость есть  - люди, состоящие в браке, имеют меньшую долю должников, по сравнению с теми, кто не в браке или в гражданском браке. Так же низкая доля должников среди тех, кто в разводе и тех, кто вдовец / вдова, что больше связано с возрастом.

3) Итак, между уровнем дохода и возвратом кредита в срок не наблюдается строгой и линейной зависимости.

4) Цели кредита влияют на его возврат в срок. Среди кредитов на свадьбу и на недвижимость 7,9% и 7,2% должников соответственно. Среди кредитов на образование и автомобиль 9,2% и 9,3% должников соответственно. Это говорит о том, что семейные люди берущие кредит на благо семьи являются более надёжными заёмщиками.